benchmark/cublas/blas1/asum.cpp

```diff
@@ -140,8 +135,11 @@ void register_benchmark(blas_benchmark::Args& args,
                          index_t size, bool* success) {
       run<scalar_t>(st, cuda_handle_ptr, size, success);
     };
-    benchmark::RegisterBenchmark(get_name<scalar_t>(size).c_str(), BM_lambda,
-                                 cuda_handle_ptr, size, success)
+    benchmark::RegisterBenchmark(
+        blas_benchmark::utils::get_name<benchmark_op, scalar_t>(
+            size, blas_benchmark::utils::MEM_TYPE_USM)
```

Comment:
Thanks for your comment. 

I used USM to address this comment from @muhammad-tanvir-1211:

> This argument can be set to usm for other backends and buffer for the SYCL-BLAS backend for now.

I think `MEM_TYPE_BUFFER` refers to use `sycl::buffer`, which only makes sense in the SYCL_BLAS backend.

Reply:
Yes, true. This applies just to the SYCL-BLAS benchmark.

Probably, we should introduce something similar to `MEM_TYPE_GLOBAL_POINTER` for `cublas` and `rocblas` for avoid confusion.

Reply:
Yes, thanks makes sense. Thanks for bringing it up. 

I can look into that as a separate PRs.

Reply:
Usm hear me as pointer approach. We don't need separate enum.

Reply:
As you prefer @mehdi-goli. I raised this point for stating clearly that we are not using any form of unified memory in `cublas` and `rocblas`, i.e., we use `{cuda|hip}Malloc` and not  `{cuda|hip}MallocManaged`.

Reply:
That is exactly the same for the USM in SYCL-BLAS. We are not using the System level shared memory, we are using Device USM, which requires explicit copy of data between device and host, we are not using the equivalent of `cudaMallocManaged` for SYLC in SYCL-BLAS.

Reply:
Exactly, but we call it USM because the SYCL specification does it, even if we are using `malloc_device`.

Reply:
Yes that is the naming SYCL uses for their pointer. So since the functionality here for both cases of CUDA and SYCL are the same, we should stick to SYCL naming as there is no different feature used here, so there is no need to different types of usage applied here.